tests/openassetio-core/trait/TraitBaseTest.cpp

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 The Foundry Visionmongers Ltd
+#include <openassetio/trait/TraitBase.hpp>
+#include <openassetio/typedefs.hpp>
+
+#include <catch2/catch.hpp>
+#include "openassetio/TraitsData.hpp"
+
+namespace {
+
+using openassetio::Str;
+using openassetio::TraitsData;
+using openassetio::TraitsDataPtr;
+using openassetio::trait::TraitBase;
+using openassetio::trait::TraitId;
+using openassetio::trait::TraitPropertyStatus;
+using openassetio::trait::property::Key;
+
+// TraitBase can't be tested directly, so we derive a test trait.
+struct TestTrait : TraitBase<TestTrait> {
+ private:
+  static inline const Key kSomeProperty{"some property"};
+
+ public:
+  static inline const TraitId kId{"test"};
+
+  using TraitBase<TestTrait>::TraitBase;
+
+  [[nodiscard]] TraitPropertyStatus getSomeProperty(Str* out) const {
+    return getTraitProperty(out, kId, kSomeProperty);
+  }
+
+  void setSomeProperty(const Str& value) { data()->setTraitProperty(kId, kSomeProperty, value); }
+
+  // Helpers for tests
+
+  [[nodiscard]] TraitsDataPtr& dataPtr() { return data(); }
+  [[nodiscard]] const TraitsDataPtr& dataPtr() const { return data(); }
+};
+
+SCENARIO("Retrieving the undelying data") {
+  GIVEN("Some known traits data") {
+    TraitsDataPtr data = std::make_shared<TraitsData>();
```

Comment:
[consider] We have our own alias of `make_shared` now (last PR merged), `openassetio::makeShared`.  Would be good to dogfood it in our tests.

Reply:
Ah yes, forgot that you wouldn't automatically get the latest changes :roll_eyes: . Yeah happy for you to do that.

Reply:
Switched [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/373/commits/db1985607e43c8b83ab4e94c66f8f38b40e2c23a).